### ISSUE TITLE
redundant_clone: false negative with unneeded ref

### ISSUE THREAD

**Issue by matthiaskrgr** [rust-lang/rust-clippy#7227]
Lint name:

I tried this code:

```rust
fn opt_string() -> Option<String> {
    Some(String::from("hello world"))
}

fn main() {
    let s: String = match opt_string() {
        Some(ref string) => string.clone(), // we take a ref here so we have to clone()
        // but actually  opt_string() returns an owned value and we could consume it
        _ => String::new(),
    };

    // this could be:

    let _s_better: String = match opt_string()  {
        Some(string) => string,
        _ => String::new(),
    };
    
    assert_eq!("hello world".to_string(), s);
}
```

### Meta

- `cargo clippy -V`: `clippy 0.1.54 (8cf990c 2021-05-15)`
- `rustc -Vv`:
  ```
rustc 1.54.0-nightly (8cf990c9b 2021-05-15)
binary: rustc
commit-hash: 8cf990c9b5c59f25c806fad9f4466f9d6509bbea
commit-date: 2021-05-15
host: x86_64-unknown-linux-gnu
release: 1.54.0-nightly
LLVM version: 12.0.1
  ```
